my-app/src/components/messageList.js

```diff
@@ -1,33 +1,48 @@
 import React, { Component } from 'react';
 import Message from './message';
+import slugify from '../utils/slugify';
+import Header from './Header'
 
 class MessageList extends Component {
+  constructor(props){
+    super(props)
+    const channelName = this.props.data.location.data.channelName;
+    const user = this.props.data.username;
+    const slug = slugify(channelName);
+    this.state = {
+      newMessage: "",
+      channelName: this.props.data.location.data.channelName,
+      user: this.props.data.username,
```

Comment:
```
 user: user
```

we have already assigned this to user on line 10 `const user = this.props.data.username;`